app/models/expert_subject.rb

```diff
@@ -78,8 +78,8 @@ class ExpertSubject < ApplicationRecord
 
   scope :in_company_registres, -> (company) do
     klass = self
-    klass = klass.not_of_institution(Institution.find_by(slug: 'cma')) if company.inscrit_rcs && !company.inscrit_rm
-    klass = klass.not_of_institution(Institution.find_by(slug: 'cci')) if !company.inscrit_rcs && company.inscrit_rm
+    klass = klass.not_of_institution(Institution.find_by(slug: 'cma')) if (company.inscrit_rcs || company.activite_liberale) && !company.inscrit_rm
+    klass = klass.not_of_institution(Institution.find_by(slug: 'cci')) if (company.inscrit_rm  || company.activite_liberale) && !company.inscrit_rcs
```

Comment:
et ici un truc comme ça 
```suggestion
    klass = klass.not_of_institution(Institution.find_by(slug: 'cci')) unless company.inscrit_rcs
```